src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/FhirJsonInputFormatter.cs

```diff
@@ -56,6 +56,25 @@ public override async Task<InputFormatterResult> ReadRequestBodyAsync(InputForma
 
             var request = context.HttpContext.Request;
 
+            var parserToUse = _parser;
+
+            if (request.Method == "POST"
+               && request.Path.Value.EndsWith("/", System.StringComparison.OrdinalIgnoreCase))
+            {
+                var newsettings = _parser.Settings.Clone();
+                newsettings.AllowUnrecognizedEnums = true;
+                newsettings.AcceptUnknownMembers = true;
+                newsettings.ExceptionHandler = (source, args) =>
+                {
+                    context.ModelState.TryAddModelError(string.Empty, string.Format(Api.Resources.ParsingError, args.Message));
+                };
+                #pragma warning disable CS0618 // Type or member is obsolete
```

Comment:
Why are we using an obsolete type?

Reply:
I am trying to use current settings as it is and add additional settings.  In our existing code during startup this parser is initialized with a setting (TruncateDateTimeToDate) that was marked as Obsolete and is there only for backward compatibility.
FhirModule.cs file line #44 to #46.

To avoid changing existing behavior, I need to bring in those settings along with new settings I added. Please let me know if there is any other way. Thanks.

Reply:
Sounds good. Could you add a comment to the code to explain this reasoning? I feel we should document when we are suppressing these rules so we can reevaluate them later.

Reply:
Added comments.